compute timespan

pkitszel opened the pull request.

pkitszel committed:
ml-stat: compute timespan for pull requests

Compute timespan for pull requests (for thread really), and print
stats per submitter.

Signed-off-by: Przemek Kitszel <przemyslaw.kitszel@intel.com>

pkitszel committed:
ml-stat: timespan: don't count late replies

Cut out long tail of late replies to given thread, to don't bump timespan
when someone replies to the thread that was silent for long time.

Done via simple heuristic.

Signed-off-by: Przemek Kitszel <przemyslaw.kitszel@intel.com>

pkitszel commented on ml-stat.py:
```diff
@@ -229,6 +241,25 @@ def authors(self, mapping):
+                    break
+            last = d
+
+        # round up to one day
+        return max(last - begin, datetime.timedelta(days=1))
+
```
perhaps we should add 1 day instead of just rounding up, to somehow account for the break between iterations

kuba-moo replied:
Yeah, tricky..
Another tricky thing is weekends.

kuba-moo commented on ml-stat.py:
```diff
@@ -82,8 +82,8 @@ def is_patch(self):
-        return 'pull req' in self._subject
-        return 'pull req' in self._subject or 'pull-req' in self._subject
+        return 'pull req' in self._subject or 'pull-req' in self._subject or \
+               'Intel Wired LAN Driver' in self._subject
```
This feels a little too much like a temporary hack, not sure if we should be putting this in the code going forward. Presumably Jake can correct his formatting instead?

kuba-moo commented on ml-stat.py:
```diff
@@ -106,31 +106,36 @@ def __init__(self, msg):
+                self._review = True
+            if l.startswith('pw-bot:'):
+                self._pw_cmd = True
+        self._body_processed = True
```
Shouldn't this be sooner? If we get an exception because email is funny we will get it next time, too

mszycik commented on ml-stat.py:
```diff
@@ -229,6 +241,20 @@ def authors(self, mapping):
+        for msg in self.msgs:
+            d = msg.date()
+            if msg.is_pwbot_command() or msg.is_pwbot_accept():
+                    last = d
+                    break
```
Indentation (does this even work?)

pkitszel committed:
ml-stat: exclude weekends from PR timespan

Exclude weekend days from PR timespan calculation.

Signed-off-by: Przemek Kitszel <przemyslaw.kitszel@intel.com>

pkitszel commented:
Changes since v1:
added weekend removal from timespan
dropped custom logic for IWL PRs
moved `self._body_processed = True` to be much earlier
whitespace fixed
